pkg/azure/provider/helpers/driver.go

```diff
@@ -428,24 +428,29 @@ func createNICTags(tags map[string]string) map[string]*string {
 // 4. If the agreement has not been accepted yet then it will accept the agreement and update the agreement. If that fails then it will return an error.
 func ProcessVMImageConfiguration(ctx context.Context, factory access.Factory, connectConfig access.ConnectConfig, providerSpec api.AzureProviderSpec, vmName string) (imgRef armcompute.ImageReference, plan *armcompute.Plan, err error) {
 	imgRef = getImageReference(providerSpec)
-	isMarketPlaceImage := providerSpec.Properties.StorageProfile.ImageReference.URN != nil
-	if isMarketPlaceImage {
-		var vmImage *armcompute.VirtualMachineImage
-		vmImage, err = getVirtualMachineImage(ctx, factory, connectConfig, providerSpec.Location, imgRef)
+
+	shouldCheckMarketplaceImage := providerSpec.Properties.StorageProfile.ImageReference.URN != nil && !providerSpec.Properties.StorageProfile.ImageReference.SkipMarketplaceAgreement
```

Comment:
Add a unit test for this skip agreement check as well.

Reply:
@rishabh-11 added both tests -> Could you PTAL ?